tests/integration/test_cogs_rates.py

```diff
@@ -21,55 +32,126 @@ def rates_cog(sample_bot):
+        title="ARK's Official server rates have just been updated!",
+        color=0x63BCC3,
     )
+    sample_embed.set_image(url=EMBED_IMAGE)
 
+    # First request; no diff. First time seeing rates
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().nothing()
 
-@pytest.mark.asyncio
-async def test_rates_compare_posted_rates_bad_fetch(rates_cog):
-    output_path = rates_cog.output_paths[0]
+    # 2nd request; no diff. Rates are now stable since they've been observed 2 times in a row
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().nothing()
 
-    with pytest.raises(RatesFetchError):
-        rates_diff = await rates_cog.compare_posted_rates(
-            "http://localhost:5000/bogus-url.txt", output_path
+    # 3rd request; still no diff. No change to stable rates
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().nothing()
+
+    # 4th request; new rates found, but not yet stable
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().nothing()
+
+    # 5th request; no diff from previous, which means new rates are stable. Different from previous stable rates, so embed is published
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().embed(embed=sample_embed)
+
+    # Only one embed was published; queue is empty after consuming message above
+    assert dpytest.verify().message().nothing()
+
+
+@pytest.mark.parametrize(
+    "sequential_handler",
+    [
+        (
+            SAMPLE_DYNAMICCONFIG_2,
+            SAMPLE_DYNAMICCONFIG_2,
+            SAMPLE_DYNAMICCONFIG_1,
+            SAMPLE_DYNAMICCONFIG_1,
         )
+    ],
+    indirect=True,
+)
+@pytest.mark.asyncio
+async def test_compare_and_notify_all_reverse(rates_cog):
+    sample_embed = discord.Embed(
+        description="3 × Taming\n**3** × Harvesting\n3 × XP\n**0.6** × Mating Interval\n3 × Maturation\n3 × Hatching\n0.6 × Cuddle Interval\n3 × Imprinting\n**1.5** × Hexagon Reward",
+        title="ARK's Official server rates have just been updated!",
+        color=0x63BCC3,
+    )
+    sample_embed.set_image(url=EMBED_IMAGE)
+
+    # First request; no diff. First time seeing rates
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().nothing()
+
+    # 2nd request; no diff. Rates are now stable since they've been observed 2 times in a row
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().nothing()
+
+    # 3rd request; still no diff. No change to stable rates
+    await rates_cog.compare_and_notify_all()
+    assert dpytest.verify().message().nothing()
+
+    # 4th request; no diff from previous, which means new rates are stable. Different from previous stable rates, so embed is published
+    await rates_cog.compare_and_notify_all()
+    print(rates_cog.last_rates[0].to_dict())
```

Comment:
Note when the `dpytest.verify()` step fails (because the message doesn't exist or doesn't match), pytest doesn't really write much to the terminal that is useful. Thankfully pytest prints anything it captured in stdout whenever a test fails. So this is one way you could help troubleshoot why a `dpytest` test function failed.